CI/Dockerfile

```diff
@@ -33,63 +54,207 @@ RUN apt-get -y update; \
 
-ARG MOAB=5.3.0
-ENV MOAB_VERSION ${MOAB}
-#TODO move sh file contents into this Dockerfile
-RUN if [ "${MOAB_VERSION}" != "master" ] && [ "${MOAB_VERSION}" != "develop" ]; then \
-        /root/etc/CI/docker/build_moab.sh; \
-    fi;
-    
\ No newline at end of file
+
+RUN mkdir -p ${moab_build_dir}/build && \
+    cd ${moab_build_dir} && \
+    git clone -b ${MOAB_BRANCH} --depth 1 https://bitbucket.org/fathomteam/moab && \
+    cd build && \
+    cmake ../moab -DCMAKE_INSTALL_RPATH=${hdf5_install_dir}/lib:${moab_install_dir}/lib \
+                  -DENABLE_HDF5=ON -DHDF5_ROOT=${hdf5_install_dir} \
+                  -DCMAKE_INSTALL_PREFIX=${moab_install_dir} \
+                  -DCMAKE_CXX_COMPILER=${CXX} \
+                  -DCMAKE_C_COMPILER=${CC} \
+                  -DENABLE_BLASLAPACK=OFF \
+                  -DBUILD_SHARED_LIBS=ON \
+                  -DENABLE_FORTRAN=OFF \
+                  -DENABLE_PYMOAB=ON && \
+    make -j${ci_jobs} && \
+    make install && \
+    cd && \
+    rm -rf ${moab_build_dir}
+
+ENV double_down_build_dir=${build_dir}/double-down
+ENV double_down_install_dir=${install_dir}/double-down
+
+# Clone and install Double-Down
+# performed after moab install as double-down requires moab
+RUN mkdir -p ${double_down_build_dir}/build && \
+    cd ${double_down_build_dir} && \
+    git clone --shallow-submodules --single-branch --branch v1.0.0 --depth 1 https://github.com/pshriwise/double-down.git && \
+    cd build && \
+    cmake ../double-down -DMOAB_DIR=${moab_install_dir} \
+                         -DCMAKE_INSTALL_PREFIX=${double_down_install_dir} \
+                         -DEMBREE_DIR=${embree_install_dir} && \
+    make -j${ci_jobs} && \
+    make -j${ci_jobs} install
+
+
+FROM moab as dagmc
+
+# accessing gloabl ARGs in build stage
+ARG install_dir
+ARG build_dir
+ARG CXX
+ARG CC
+
+ENV dagmc_build_dir=${build_dir}/dagmc
+ENV dagmc_install_dir=${install_dir}/dagmc
+
+# copies the enitre git repo into the dockerfile to ensure the submodules can also be found
+RUN mkdir -p ${dagmc_build_dir}/build
+COPY . ${dagmc_build_dir}
+RUN cd  ${dagmc_build_dir} && git submodule update --init
+
+RUN mkdir -p ${dagmc_build_dir}/build && \
+    cd ${dagmc_build_dir} && \
+    cd build && \
+    cmake ../${dagmc} -DMOAB_DIR=${moab_install_dir} \
+                      -DBUILD_GEANT4=ON \
+                      -DGEANT4_DIR=${geant4_install_dir} \
+                      -DBUILD_CI_TESTS=ON \
+                      # reg tests disabled see issue https://github.com/svalinn/DAGMC/issues/858
+                      -DBUILD_MW_REG_TESTS=OFF \
+                      -DBUILD_STATIC_EXE=OFF \
+                      -DBUILD_STATIC_LIBS=OFF \
+                      -DCMAKE_C_COMPILER=${CC} \
+                      -DCMAKE_CXX_COMPILER=${CXX} \
+                      -DCMAKE_Fortran_COMPILER=gfortran \
+                      -DCMAKE_INSTALL_PREFIX=${dagmc_install_dir} \
+                      -DDOUBLE_DOWN=${double_down} \
```

Comment:
not sure where/if `double_down` is set?  I think it should always be `ON` given our current testing plan

Reply:
This is set to `OFF` by default, but we always build double down anyway...  probably OK like this.